### PR TITLE
error handling within interceptors.ts

### PR DESCRIPTION
**Pull request by paddybasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/lib/interceptors.ts [hunk 1]
```diff
@@ -54,6 +54,9 @@ export function errorInterceptor(error) {
     let data = valueOrNull(error, 'response.data.details')
     if (!data) {
         data = valueOrNull(error, 'response.status') ? JSON.stringify(error.response.data, null, 2) : null
+        if (!data) {
+            data = error
+        }
         logger.error(`Error on ${error.config.method.toUpperCase()} to ${url} in (${error.duration}) - ${error} \n
```
Comment (timja) [inline api/lib/interceptors.ts:60]: is it guaranteed that `error.config.method` will always be set?

Comment (paddybasi) [reply api/lib/interceptors.ts:60]: from what I can see yes, have done a debug session (changed the ccd url etc). This comes from axios directly

Comment (timja) [reply api/lib/interceptors.ts:60]: are all errors going to be from API calls though? it's probably good enough for now, but possibly needs re-visiting later

Comment (paddybasi) [reply api/lib/interceptors.ts:60]: this interceptor will only get called for any failed axios requests so hopefully good enough for now. But agreed it won't catch any TypeErrors etc

**Pull request merged**